TLM/TLM/Custom/PathFinding/CustomPathFind.cs

```diff
@@ -4000,7 +4012,9 @@ private struct BufferItem {
                 laneInfo,
                 VehicleRestrictionsMode.Configured);
 
-            return (allowedTypes & queueItem_.vehicleType) != ExtVehicleType.None;
+            return (allowedTypes & queueItem_.vehicleType) != ExtVehicleType.None ||
+                   startLaneA_ == laneId || startLaneB_ == laneId ||
+                   endLaneA_ == laneId || endLaneB_ == laneId;
```

Comment:
I suspect this will solve a load of weird problems that we've had in the past :)

Reply:
Yup, those related with vehicle restrictions for sure